### PR TITLE
feat: build collection-tutorial-list for collection views

### PR DESCRIPTION
**Pull request by zchsh**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-zscollection-tutorial-list-hashicorp.vercel.app/waypoint/tutorials/get-started-docker) 🔎
- [Asana task](https://app.asana.com/0/1202001387788447/1202149167439225/f) 🎟️

## 🗒️ What

Implements a `CollectionTutorialList` component, and uses it on all collection views.

## 🤷 Why

To continue building out our `collection-view` in dev-dot.

## 🛠️ How

- Uses existing `tutorial-card` component in a `display: grid` layout
- Accepts an `isOrdered` prop, as different collection types have different grid characteristics

## 📸 Design Screenshots

<details>

<summary>Ordered CollectionTutorialList</summary>

### Design spec

TO DO

### Implementation

TO DO

</details>

<details>

<summary>Unordered CollectionTutorialList</summary>

### Design spec

TO DO

### Implementation

TO DO

</details>

## 🧪 Testing

- [ ] Visit an ordered collection, such as [/waypoint/tutorials/get-started-docker][ordered-collection]
    - [ ] `CollectionTutorialList` should render in one column at all viewport sizes
- [ ] Visit an unordered collection, such as [/vault/tutorials/secrets-management][unordered-collection]
    - [ ] `CollectionTutorialList` should render in one column on mobile, and max out at 2 columns on larger viewports

> Note: you can also review [in Storybook][storybook], which I found helpful for testing responsive behaviour.

## 💭 Anything else?

Not at the moment!

[ordered-collection]: https://dev-portal-git-zscollection-tutorial-list-hashicorp.vercel.app/waypoint/tutorials/get-started-docker
[unordered-collection]: https://dev-portal-git-zscollection-tutorial-list-hashicorp.vercel.app/vault/tutorials/secrets-management
[storybook]: https://dev-portal-with-storybook.vercel.app/?path=/story/components-collectiontutoriallist--playground

### CODE REVIEW

File: src/components/tutorial-card/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { TutorialCardPropsWithId } from './types'
+
+export function formatTutorialCard(
+  tutorial: ClientTutorialLite,
+  collectionSlug: string
+): TutorialCardPropsWithId {
+  return {
+    id: tutorial.id,
+    description: tutorial.description,
+    duration: `${tutorial.readTime}min`,
```
Comment (zchsh) [inline src/components/tutorial-card/helpers.ts:15]: @kendallstrautman I guess I lied in #355, `duration` has ended up still being a `string` in this PR 😅 I liked the existing `formatCollectionCard` pattern you'd set up for `FeaturedInCollections`, so I swiped it for tutorial cards. Let me know what you think 🙇‍♂️

File: src/views/collection-view/components/collection-tutorial-list/collection-tutorial-list.module.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  list-style: none;
+  margin: 0;
+  padding: 16px;
+
+  @media (--dev-dot-tablet-up) {
+    /**
+     * On larger viewports, where our content area reaches up to 780px width,
+     * cards are meant to be in two columns.
+     * 
+     * Note: we could explicitly set 2 columns here instead - it would achieve
```
Comment (zchsh) [inline src/views/collection-view/components/collection-tutorial-list/collection-tutorial-list.module.css:28]: I don't know if this merits such a long comment, but felt it was worth explaining the rationale.

File: src/views/collection-view/index.tsx [hunk 3]
```diff
@@ -32,18 +34,12 @@ function CollectionView({
         numTutorials={tutorials.length}
       />
       <h2 id={layoutProps.headings[1].slug}>Tutorials</h2>
-      <ol>
```
Comment (zchsh) [inline src/views/collection-view/index.tsx:35]: Just wanna say it felt really smooth switching from the placeholder component to the polished one 😍 I really appreciate the excellent stubbing work 💯

File: src/views/collection-view/components/collection-tutorial-list/collection-tutorial-list.module.css (hunk 2 above)
Comment (kendallstrautman) [reply src/views/collection-view/components/collection-tutorial-list/collection-tutorial-list.module.css:28]: I think this is great, thanks for explaining your rationale

File: src/views/collection-view/components/collection-tutorial-list/index.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import classNames from 'classnames'
+import TutorialCard from 'components/tutorial-card'
+import { TutorialCardPropsWithId } from 'components/tutorial-card/types'
+import s from './collection-tutorial-list.module.css'
+
+function CollectionTutorialList({
+  tutorials,
+  isOrdered,
+}: {
```
Comment (kendallstrautman) [inline src/views/collection-view/components/collection-tutorial-list/index.tsx:9]: Total style nit, but might be worth extracting into own interface.

File: src/views/collection-view/components/collection-tutorial-list/index.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  tutorials: TutorialCardPropsWithId[]
+  isOrdered: boolean
+}) {
+  const ListRoot = isOrdered ? 'ol' : 'ul'
```
Comment (kendallstrautman) [inline src/views/collection-view/components/collection-tutorial-list/index.tsx:13]: Oh that's interesting, I didn't realize you could define the element just via a string like this! Cool 😎

File: src/views/collection-view/index.tsx [hunk 6]
```diff
@@ -32,18 +34,12 @@ function CollectionView({
         numTutorials={tutorials.length}
       />
       <h2 id={layoutProps.headings[1].slug}>Tutorials</h2>
```
Comment (kendallstrautman) [inline src/views/collection-view/index.tsx:36]: Might not be worth tackling in this PR, but we should figure out what to do with the Tutorials heading here.

**Review by kendallstrautman — APPROVED** [review]
Super clean! Besides that extra duration formatting thing, this is good to go 🚀

File: src/views/collection-view/index.tsx (hunk 6 above)
Comment (kendallstrautman) [reply src/views/collection-view/index.tsx:36]: Made a task so we don't forget - https://app.asana.com/0/1202001387788447/1202168098250852

File: src/views/collection-view/components/collection-tutorial-list/index.tsx (hunk 4 above)
Comment (zchsh) [reply src/views/collection-view/components/collection-tutorial-list/index.tsx]: ✅ Great call - done!

**Pull request merged**